cloud_experts_tutorials/cloud-experts-deploying-application/cloud-experts-deploying-application-integrating-aws.adoc

```diff
@@ -0,0 +1,391 @@
++
+[source,terminal]
+----
+$ export OSTOY_NAMESPACE=$(oc config view --minify -o 'jsonpath={..namespace}')
+----
+
+== Creating an AWS IAM role
+
+. Get your AWS account ID by running the following command:
++
+[source,terminal]
+----
+$ export AWS_ACCOUNT_ID=$(aws sts get-caller-identity --query Account --output text)
+----
+. Get the OIDC provider by running the following command, replacing `<cluster-name\>` with the name of your cluster:
++
+[source,terminal]
+----
+$ export OIDC_PROVIDER=$(rosa describe cluster -c <cluster-name> -o yaml | awk '/oidc_endpoint_url/ {print $2}' | cut -d '/' -f 3,4)
+----
++
+. Create the trust policy file by running the following command:
++
+[source,terminal]
+----
+$ cat <<EOF > ./ostoy-sa-trust.json
+{
+"Version": "2012-10-17",
+"Statement": [
+{
+"Effect": "Allow",
+"Principal": {
+"Federated": "arn:aws:iam::${AWS_ACCOUNT_ID}:oidc-provider/${OIDC_PROVIDER}"
+},
+"Action": "sts:AssumeRoleWithWebIdentity",
+"Condition": {
+"StringEquals": {
+"${OIDC_PROVIDER}:sub": "system:serviceaccount:${OSTOY_NAMESPACE}:ostoy-sa"
+}
+}
+}
+]
+}
+EOF
+----
++
+. Create the AWS IAM role to be used with your service account by running the following command:
++
+[source,terminal]
+----
+$ aws iam create-role --role-name "ostoy-sa-role" --assume-role-policy-document file://ostoy-sa-trust.json
+----
+
+== Attaching the S3 policy to the IAM role
+
+. Get the S3 full access policy ARN by running the following command:
++
+[source,terminal]
+----
+$ export POLICY_ARN=$(aws iam list-policies --query 'Policies[?PolicyName==`AmazonS3FullAccess`].Arn' --output text)
+----
++
+. Attach the policy to the AWS IAM role by running the following command:
++
+[source,terminal]
+----
+$ aws iam attach-role-policy --role-name "ostoy-sa-role" --policy-arn "${POLICY_ARN}"
+----
+
+== Creating the service account for your pod
+
+. Get the ARN for the AWS IAM role we created so that it will be included as an annotation when you create your service account by running the following command:
++
+[source,terminal]
+----
+$ export APP_IAM_ROLE_ARN=$(aws iam get-role --role-name=ostoy-sa-role --query Role.Arn --output text)
+----
++
+. Create the service account by running the following command:
++
+[source,terminal]
+----
+$ cat <<EOF | oc apply -f -
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+name: ostoy-sa
+namespace: ${OSTOY_NAMESPACE}
+annotations:
+eks.amazonaws.com/role-arn: "$APP_IAM_ROLE_ARN"
+EOF
+----
++
+[IMPORTANT]
+====
+Do not change the name of the service account from "ostoy-sa" or you will have to change the trust relationship for the AWS IAM role.
+====
++
+. Grant the service account the `restricted` role by running the following command:
++
+[source,terminal]
+----
+$ oc adm policy add-scc-to-user restricted system:serviceaccount:${OSTOY_NAMESPACE}:ostoy-sa
+----
++
+. Confirm that the previous step was successful by running the following command:
```

Comment:
this is not specific to the previous step of granting the restricted role but rather calling the reader to confirm that the annotation was successful